oeps/processes/oep-0056-architectural-advisory-process.rst

```diff
@@ -0,0 +1,197 @@
+
+Under-utilized ADRs
+-------------------
+
+We adopted ADRs in :doc:`../best-practices/oep-0019-bp-developer-documentation` several years ago as way of recording techincal decisions and co-locating them with their corresponding code.  OEP-19 is otherwise unopinionated about how those ADRs get used and reviewed, or when in our engineering process writing them should occur.  This has led to ambiguity and ambivalence around ADRs. Often they're an afterthought, and serve only as grudging documentation of a decision after it has been made.
+
+We believe that ADRs - and more importantly, the conversations that happen around them - can be so much more.  We can use these documents and conversations as a vehicle for decentralizing and scaling our decision making process, making more informed decisions, democratizing access to domain knowledge, elevating our architectural discourse, and recording our decisions for posterity with additional contextual information that's often missing today.
+
+Finding expertise is difficult
+------------------------------
+
+Related to the above, historically it is also very difficult to find experts and stakeholders in a given domain.  Generally this happens via word of mouth, sometimes not at all, or occasionally by engineers realizing a vacuum of expertise exists and *they are now the expert!*
+
+This difficulty is compounded by the decentralized and global nature of the Open edX community.
+
+Atrophy of architectural principles and tech radar
+--------------------------------------------------
+
+An important component of our ability to make architectural decisions that are aligned across teams and organizations is measuring and vetting those decisions against a set of `shared architectural principles`_ and technologies in our `Tech Radar`_.  Today, we lack organized maintenance of our principles and radar, nor do we have an appropriate forum in which to discuss and socialize them.
+
+Specification
+=============
+
+We believe that we can create a healthy and empowering dialogue around our architectural process and mitigate the above concerns by clarifying the spirit and process around the creation and review of ADRs.
+
+We will adjust our current architectural process by:
+
+- Formalizing and clarifying an **Architectural Advisory Process** in which we conversationally provide input on and promptly review the status of ADRs, as well as review updates to our architectural principles and tech radar, with the goal of enabling more effective, informed decision making.
+- Improve the discoverability of stakeholders and experts via a **decision maker's checklist**.
+- Set decision makers up for success by using the Architectural Advisory Process to **vet changes to our architectural principles and tech radar**.  Both are crucial inputs into the decision making process by helping to align our community on our architectural values and technology choices.
+
```

Comment:
Right now Github Issues are "optional" but my experience of this process so far is that the biggest roadblock to discoverability is the delay in creating a centralized artifact due to the inhibition around draft ADRs. I would like to see at least a line in here that asserts our intention to create a public, centralized, discoverable artifact (whether it is an Issue or a draft ADR in a PR) well in advance of a decision and preferably even before possible solutions have been identified. I want to see **open problems** which are discoverable and specific (not distributed across stale wikis and slack threads in the form of institutional knowledge or general concerns).

PS. I will also acknowledge that some of the discoverability is impacted by the "What repo should this go in" question, but I think that is a solvable problem.

PPS. This has progressed so much since I last read it! You should all be very proud! And because there are a lot of changes and a lot of open comments I apologize in advance if I'm repeating some of them.

Reply:
My reasoning in calling it "optional" was that I don't think it's always necessary, not because it's contentious.  My gut says _making_ it necessary could increase cognitive load/friction in some cases.